docs/design/features/dotnet-pgo.md

```diff
@@ -0,0 +1,88 @@
+# dotnet-pgo Spec
+Utilize trace data for improving application performance
+
+NOTE: This documentation page contains information on some features that are still work-in-progress.
+
+## Intro
+
+The dotnet-pgo tool is a cross-platform CLI global tool that enables conversion of traces of .NET Core applications collected via dotnet-trace, ETW, perfview, perfcollect, LTTNG to be used to improve the performance of an application or library.
+
+## Installing dotnet-pgo
+
+The first step is to install the dotnet-pgo CLI global tool.
+
+```cmd
+$ dotnet tool install --global dotnet-pgo
+You can invoke the tool using the following command: dotnet-pgo
+Tool 'dotnet-pgo' (version '6.0.47001') was successfully installed.
+```
+
+## Using dotnet-pgo to optimize an application
+
+In order to use dotnet-pgo you will need to:
+
+1. Collect traces of you application with at least the following trace setting. `Microsoft-Windows-DotNETRuntime:0x1C000080018:4`
+2. Run the dotnet-pgo tool specifying the trace file collected above to create a mibc file.
+3. Pass the mibc file to a `dotnet publish` command. Specify the file via the `ReadyToRunOptimizationData` property. Multiple files may be passed separated by `;` characters.
+
+## Command line reference for dotnet-pgo
+
+dotnet-pgo supports two commands. The `create-mibc` is used to convert a trace into a Mibc file for usage by the .NET build. The `merge` command is used to merge multiple mibc files together to reduce usage complexity when an application uses several test scenarios.
+
+```
+usage: dotnet-pgo <command> [<args>]
+
+    create-mibc    Transform a trace file into a Mibc profile data file.
+    merge          Merge multiple Mibc profile data files into one file.
+```
+
+### `create-mibc` command
+Transform a trace file into a Mibc profile data file.
+```
+usage: dotnet-pgo create-mibc --trace-file <arg> --output-file-name <arg> [--pid <arg>] [--process-name [arg]] [--clr-instance-id <arg>] [--reference <arg>...] [--exclude-events-before <arg>]
+                  [--exclude-events-after <arg>] [--verbosity <arg>] [--compressed] [-h]
+
+    --trace-file <arg>               Specify the trace file to be parsed.
+    --output-file-name <arg>         Specify the output filename to be created.
```

Comment:
Rename to --output and also support -o to be more similar to the dotnet-trace tool.